lib/ferrum/page.rb

```diff
@@ -149,17 +153,39 @@ def close_connection
     # @param [Boolean] mobile whether to emulate mobile device
     #
     def set_viewport(width:, height:, scale_factor: 0, mobile: false)
-      command(
-        "Emulation.setDeviceMetricsOverride",
-        slowmoable: true,
-        width: width,
-        height: height,
-        deviceScaleFactor: scale_factor,
-        mobile: mobile
-      )
+      if mobile
+        command(
+          "Emulation.setTouchEmulationEnabled",
+          enabled: true,
+          maxTouchPoints: 1
+        )
+
+        command(
```

Comment:
I'm uncertain about this.  Ideally the caller would always specify a mobile height, width, and scale factor.  But in principle that doesn't always happen, so I decided to override them here.  

Perhaps we should raise an exception if a height and width other than `0` are passed along with `mobile: true`?

Reply:
Why setting it hard in the first place?

Reply:
It isn't, but it's also surprising - if we didn't hard-code the resolution here, users would have to know:

1. That they _need_ to set the resolution at the same time as enabling mobile, and
2. What the right resolution is for mobile devices.

Hence my suggestion that perhaps I should make this raise an exception if the user also passes height or width?  That way it should be really clear.

Reply:
Have made it raise an exception now, so the behaviour is:

```
context "when a width is passed" do
  it "raises an error" do
    expect do
      page.resize(width: 100, height: 0, mobile: true)
    end.to raise_error("specifying `mobile: true` causes automatic resizing to mobile dimensions")
    end
  end
```

Reply:
After reviewing with @DanielHeath, will *not* implement hard-coded resizing in Ferrum, but instead provide common mobile devices in Cuprite to choose from.